src/core/ext/client_config/client_channel.c

```diff
@@ -784,6 +797,8 @@ void grpc_client_channel_set_resolver(grpc_exec_ctx *exec_ctx,
     grpc_resolver_next(exec_ctx, resolver, &chand->resolver_result,
                        &chand->on_resolver_result_changed);
   }
+  chand->client_channel_factory = client_channel_factory;
+  grpc_client_channel_factory_ref(client_channel_factory);
```

Comment:
I'd suggest adding an assert over `client_channel_factory` at the beginning of the function, both as "documentation" and also to prevent funky failures if we try to unref a `NULL`.